### DISCUSSION TITLE
Barrierefreiheit / Accessibility

### DISCUSSION THREAD

**Discussion by kuhmic** [rustdesk/rustdesk#957] [Feature Request]
Hallo, 

zu aller erst einmal muss ich sagen, dass das Programm eine super alternative zu Teamviewer ist. 

Für sehende Menschen lässt sich das Programm super bedienen. Es ist einfach gehalten und grundsätzlich übersichtlich gestaltet. 

Selbst kenne ich einige Menschen, die Blind, bzw. hochgradig sehbehindert sind. Diese Menschen müssen mit einer Zusatzsoftware  (Sreenreader) arbeiten. 

Heute haben ich mit einem Freund das Programm ausprobiert. Leider kann er mit den Programmen: JAWS, NVDA, Dolphin und wie die Programme so heißen, die Inhalte nicht auslesen. 

Ich bin kein Programmierer und kenne mich mit dieser Materie nicht aus. Dennoch glaube ich, dass es nicht all zu schwer ist, das Programm auch barrierefreier zu gestalten. 

Es gibt verschiedene Schaltflechen bei denen eine Beschriftung fehlt. Durch das Fehlen können diese Flächen nicht ausgelesen werden. 

Alles was auf dem ersten Bild rot umkreist ist, sind sogenannte Schalter die eben nicht vorgelesen werden. Somit können diese auch nicht geöffnet werden. 

Auf dem zweiten Bild ist unter den Einstellungen ein Menüpunkt, der eine Beschriftung hat. Hier muss in der Programmzeile irgend ein Unterschied zu den anderen Zeilen sein. Vielleicht kann man sich an diesem orientieren und dementsprechend alles andere auch so gestalten. 

-------------------------------

Hello,

First of all, I have to say that the program is a great alternative to Teamviewer.

The program is easy to use for sighted people. It is kept simple and basically clearly laid out.

I know a few people who are blind or severely visually impaired. These people have to work with additional software (screen reader).

Today I tried the program with a friend. Unfortunately, he can't read the content with the programs: JAWS, NVDA, Dolphin and whatever the programs are called.

I am not a programmer and am not familiar with this matter. Nevertheless, I believe that it is not too difficult to make the program more barrier-free.

There are various buttons that are not labeled. Due to the absence, these areas cannot be read out.

Everything that is circled in red in the first picture are so-called switches that are not read out. This means that they cannot be opened either.

In the second picture, under the settings, there is a menu item that has a label. There must be some difference to the other lines in the program line. Maybe you can use this as a guide and design everything else accordingly.
![Bildschirmfoto 2022-07-11 um 15 48 07 PM](https://user-images.githubusercontent.com/89203419/178287197-02639220-118c-456b-86fc-147d36d81eb7.png)
![Bildschirmfoto 2022-07-11 um 16 00 01 PM](https://user-images.githubusercontent.com/89203419/178287282-f86cb15f-7159-4817-bcd3-931961994c25.png)

**Comment by rustdesk** [comment]
Thanks

**Reply by Kenforsrf** [reply]
Hello
When using Rustdesk with screen readers I have noticed similar issues. The content of fields like ID and password is read, but not the labels. This means that a blind person is not informed of what the field is used for which makes it hard to use Rustdesk. With sematics in flutter there might be possible to merge the sematic for the label and field so that the screen reader program can read both the label and the content. 

With my comment I would like to ask if there are some plans to make Rustdesk more accessible for visully impaired users?
